scripts/constituents.py

```diff
@@ -112,7 +112,7 @@ def find_variable(self, standard_name=None, source_var=None,
                 # end for
                 newdims.append(':'.join(new_dnames))
             # end for
-            var = source_var.clone({'dimensions' : newdims}, remove_intent=True,
+            var = source_var.clone({'dimensions' : newdims}, remove_intent=False,
```

Comment:
I cannot wrap my head around this change. The constituents dictionary was never intended to be a source of variables for dummy arguments. I need a clear explanation why this is necessary.

Reply:
This I did keep. I needed to know if a constituent was intent out because the framework needs to throw an error if you try to declare an intent "out" constituent at runtime (or, just any time other than "init" time)